platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/ProblemReportDetails.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.configuration.problems
+
+import org.gradle.api.internal.DocumentationRegistry
+import org.gradle.internal.cc.impl.problems.JsonSource
+import org.gradle.internal.cc.impl.problems.JsonWriter
+
+
+data class ProblemReportDetails(
+    val buildDisplayName: String?,
+    val cacheAction: String,
+    val cacheActionDescription: StructuredMessage,
+    val requestedTasks: String?,
+    val totalProblemCount: Int
+)
+
+
+class ProblemReportDetailsJsonSource(val details: ProblemReportDetails) : JsonSource {
+    override fun writeToJson(jsonWriter: JsonWriter) {
+        with(jsonWriter) {
+            property("totalProblemCount") {
+                write(details.totalProblemCount.toString())
+            }
```

Comment:
❓  Why do we have the top-level data writing logic duplicated here and in `DefaultProblemsReportCreator` ?

Reply:
It won't stay duplicated once we finalize what actual data we need to transfer.